src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -243,12 +243,15 @@ private function extractFromReflectionType(\ReflectionType $reflectionType, \Ref
 
         foreach ($reflectionType instanceof \ReflectionUnionType ? $reflectionType->getTypes() : [$reflectionType] as $type) {
             $phpTypeOrClass = $reflectionType instanceof \ReflectionNamedType ? $reflectionType->getName() : (string) $type;
+            if ('null' === $phpTypeOrClass && $reflectionType instanceof \ReflectionUnionType) {
```

Comment:
I'm not sure it's possible to have 'null' without a `ReflectionUnionType`
this means we could remove the instanceof check, and remove the comparison to 'null' below also